packages/workflow/src/records/fhir.ts

```diff
@@ -570,15 +571,35 @@ export function createRegisterTask(
   }
 }
 
-export function createArchiveTask(
+export function updateBusinessStatusInTask(
   previousTask: SavedTask,
+  status: TaskStatus
+): SavedTask {
+  return {
+    ...previousTask,
+    businessStatus: {
+      coding: [
+        {
+          system: 'http://opencrvs.org/specs/reg-status',
+          code: status
+        }
+      ]
+    }
+  }
+}
+
+function updateTaskWithCommentAndReason(
```

Comment:
I would say a better generic function which we can use for creating all the task would have an api kinda like this: `createTask(previousTask: Task, status: TaskStatus, extensions: Extension[])`. If we want to add comment/reason then we can do it in separately where needed e.g. reject, archive etc